### PR TITLE
commands: Add goto first non-whitespace char of line

### PR DESCRIPTION
**Pull request by wojciechkepka**
Closes: #163

### CODE REVIEW

File: helix-term/src/commands.rs [hunk 1]
```diff
@@ -216,6 +217,27 @@ pub fn move_line_start(cx: &mut Context) {
...
 }
 
+pub fn move_first_nonwhitespace(cx: &mut Context) {
+    let (view, doc) = cx.current();
+
+    let selection = doc.selection(view.id).transform(|range| {
+        let text = doc.text();
+        let line_idx = text.char_to_line(range.head);
+
+        let max_to = doc.text().len_chars().saturating_sub(1);
```
Comment (archseer) [inline helix-term/src/commands.rs:227]: `to`/`max_to` is unnecessary for these two functions, you calculate it but it's not used anywhere

Comment (wojciechkepka) [reply helix-term/src/commands.rs]: That's right, I forgot to clean up. Done.

File: helix-core/src/lib.rs [hunk 2]
```diff
@@ -18,16 +18,8 @@ pub mod syntax;
...
-        if ch != ' ' && ch != '\t' && ch != '\n' {
-            return Some(start);
-        }
-    }
-
-    None
+pub fn find_first_non_whitespace_char2(line: RopeSlice) -> Option<usize> {
+    line.chars().position(|ch| !ch.is_whitespace())
 }
 pub(crate) fn find_first_non_whitespace_char(text: RopeSlice, line_num: usize) -> Option<usize> {
```
Comment (archseer) [inline helix-core/src/lib.rs:24]: Can you get rid of this other function and then rename the `char2` one? It's used in a single place in indent.rs

Comment (wojciechkepka) [reply helix-core/src/lib.rs]: Done.

**Comment by pickfire** [comment]
Shouldn't this be `gi` like kakoune?

**Comment by archseer** [comment]
Discussion happened here: https://github.com/helix-editor/helix/issues/163#issuecomment-856141972

**Comment by pickfire** [comment]
I think we should have a document differences from vim and kakoune later on.

**Pull request merged**